runconfig/config.go

```diff
@@ -134,28 +134,29 @@ func NewCommand(parts ...string) *Command {
-	Image           string                // Name of the image as it was passed by the operator (eg. could be symbolic)
-	Volumes         map[string]struct{}   // List of volumes (mounts) used for the container
-	VolumeDriver    string                // Name of the volume driver used to mount volumes
-	WorkingDir      string                // Current directory (PWD) in the command will be launched
-	Entrypoint      *Entrypoint           // Entrypoint to run when starting the container
-	NetworkDisabled bool                  // Is network disabled
-	MacAddress      string                // Mac Address of the container
-	OnBuild         []string              // ONBUILD metadata that were defined on the image Dockerfile
-	Labels          map[string]string     // List of labels set to this container
+	Hostname         string                       // Hostname
+	Domainname       string                       // Domainname
+	User             string                       // User that will run the command(s) inside the container
+	AttachStdin      bool                         // Attach the standard input, makes possible user interaction
+	AttachStdout     bool                         // Attach the standard output
+	AttachStderr     bool                         // Attach the standard error
+	ExposedPorts     map[nat.Port]struct{}        // List of exposed ports
+	PublishService   string                       // Name of the network service exposed by the container
+	Tty              bool                         // Attach standard streams to a tty, including stdin if it is not closed.
+	OpenStdin        bool                         // Open stdin
+	StdinOnce        bool                         // If true, close stdin after the 1 attached client disconnects.
+	Env              []string                     // List of environment variable to set in the container
+	Cmd              *Command                     // Command to run when starting the container
+	Image            string                       // Name of the image as it was passed by the operator (eg. could be symbolic)
+	Volumes          map[string]struct{}          // List of volumes (mounts) used for the container
+	VolumeDriver     string                       // Name of the volume driver used to mount volumes
+	WorkingDir       string                       // Current directory (PWD) in the command will be launched
+	Entrypoint       *Entrypoint                  // Entrypoint to run when starting the container
+	NetworkDisabled  bool                         // Is network disabled
+	MacAddress       string                       // Mac Address of the container
+	OnBuild          []string                     // ONBUILD metadata that were defined on the image Dockerfile
+	Labels           map[string]string            // List of labels set to this container
+	AllowedBuildArgs map[string]map[string]string // list of build-time args that are allowed for expansion/substitution and passing to commands in 'run'.
```

Comment:
Ah, sure let me try by showing a build that chains three images viz. `imgA`, `imgB` and `imgC`, which are chained in that order and let's say all need a build-time arg named `foo`. I hope it brings out the UX differences. 

Following shows the build instructions with current logic of logging `ARGS` in image meta and inheritance from parent image. *Note how the ARG is just defined once in parent image and available as a trusted ARG in children images as well*:

```
# docker build -t imgA --build-arg foo=bar << EOF
FROM busybox
ARG foo
RUN <command dependent on foo>
EOF

# docker build -t imgB --build-arg foo=bar << EOF
FROM imgA
RUN <command dependent on foo>
EOF

# docker build -t imgC --build-arg foo=bar << EOF
FROM imgB
RUN <command dependent on foo>
EOF
```

Following shows the build instructions with the logic that doesn't logs `ARGS` in image meta and inheritance from parent. *Note how the ARG needs to be defined in each image's Dockerfile*:

```
# docker build -t imgA --build-arg foo=bar << EOF
FROM busybox
ARG foo
RUN <command dependent on foo>
EOF

# docker build -t imgB --build-arg foo=bar << EOF
FROM imgA
ARG foo
RUN <command dependent on foo>
EOF

# docker build -t imgC --build-arg foo=bar << EOF
FROM imgB
ARG foo
RUN <command dependent on foo>
EOF
```